api/src/main/java/org/apache/iceberg/transforms/Transforms.java

```diff
@@ -57,10 +60,26 @@ private Transforms() {
       return Identity.get(type);
     }
 
+    if (type.typeId() == Type.TypeID.TIMESTAMP) {
+      try {
+        Matcher timeZoneOffset = HAS_TIME_OFFSET.matcher(transform);
+        if (timeZoneOffset.matches()) {
+          String name = timeZoneOffset.group(1);
+          String offsetId = timeZoneOffset.group(2);
+          return TimestampTransform.get(type, name.toLowerCase(Locale.ENGLISH))
```

Comment:
`TimestampTransform.get` is responsible for handling mixed case, so there is no need to call `toLowerCase` here.